runtime/compiler/control/JITaaSCompilationThread.cpp

```diff
@@ -123,9 +124,9 @@ void handler_IProfiler_profilingSample(JITaaS::J9ClientStream *client, TR_J9VM *
    if (wholeMethodInfo)
       {
       // Serialize all the information related to this method
-      iProfiler->serializeAndSendIProfileInfoForMethod(method, comp, client);
+      iProfiler->serializeAndSendIProfileInfoForMethod(method, comp, client, abort);
       }
-   else  // Send information just for this entry
+   else if (!wholeMethodInfo || abort) // Send information just for this entry
```

Comment:
Why is this test needed? If `wholeMethodInfo` is false, `!wholeMethodInfo` is going to be true and we enter this `if` statement anyway.